src/hooks/__tests__/useChannel.test.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import PubNub from 'pubnub';
+import { cleanup, render } from '@testing-library/react';
+import { PubNubProvider } from '../../context/PubNubContext';
+import { useChannel } from '../useChannel';
+
+describe('useChannel hook', () => {
+  afterEach(cleanup);
+
+  let pubNubClient: PubNub;
+
+  beforeEach(() => {
+    if (pubNubClient) {
+      pubNubClient.unsubscribeAll();
+    }
+
+    pubNubClient = new PubNub({
+      publishKey: 'x',
+      subscribeKey: 'y',
+    });
+  });
+
+  it('should return publish and unsubscribe functions', () => {
+    const Child = () => {
+      const { publish, unsubscribe } = useChannel(['myChannel'], message => {
+        console.log(message);
```

Comment:
There are aspects of this I like.  Pairing the subscribe and unsubscribe automatically.  The ability to useState inside the listener to build state within the component.

I think this works well for some simple examples, but I wonder if you'd still use it for more complicated examples.

For example, the publish would likely be needed much lower in the component hierarchy, leading to prop drill down for it.  Or you'd be back to usePubNub to get at the publish method, in which case, what is the advantage of returning it from useChannel?